File: core/src/main/scala/com/graphbrain/db/EdgeType.scala
package com.graphbrain.db

case class EdgeType(override val id: String,
                    label: String="",
                    override val degree: Int = 0,
                    override val ts: Long = -1)
  extends Vertex(id, degree, ts) {

  def this(id: String, map: Map[String, String]) =
    this(id,
      map("label"),
      map("degree").toInt,
      map("ts").toLong)

  override def extraMap = Map("label" -> label)

  override def setId(newId: String): Vertex = copy(id=newId)

  override def setDegree(newDegree: Int): Vertex = copy(degree=newDegree)

  override def setTs(newTs: Long): Vertex = copy(ts=newTs)

  def isNegative = EdgeType.isNegative(id)
}

object EdgeType {
  def isNegative(id: String) = ID.parts(id)(0) == "neg"
}

File: core/src/main/java/com/graphbrain/db/EdgeType.java
package com.graphbrain.db;

import java.util.Map;

public class EdgeType extends Vertex {

    private String label;

    @Override
    public VertexType type() {return VertexType.EdgeType;}

    public EdgeType(String id, String label, int degree, long ts) {
        super(id, degree, ts);
        this.label = label;
    }

    public EdgeType(String id, String label) {
        this(id, label, 0, -1);
    }

    public EdgeType(String id) {
        this(id, "");
    }

    @Override
    public Vertex copy() {
        return new EdgeType(id, label, degree, ts);
    }

    public static boolean isNegative(String id) {
        return ID.parts(id)[0].equals("neg");
    }

    public boolean isNegative() {
        return isNegative(id);
    }

    public String getLabel() {
        return label;
    }
}